viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
+    
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    
+    if not user_data["watched"]:
+        return None
+    else: 
+        genre_dict = {}
+        for movie in user_data["watched"]:
+            if movie["genre"] in genre_dict:
+                genre_dict[movie["genre"]] += 1
+            else:
+                genre_dict[movie["genre"]] = 1
+        
+        popular_genre = max(genre_dict, key=genre_dict.get)
+        print(genre_dict)
+        return popular_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    unique_movies = []
+    friend_watched_movies = []
+
+    for movies in user_data["friends"]:
+        for movie in movies["watched"]:
+            friend_watched_movies.append(movie["title"])
+
+    for film in user_data["watched"]:
+        if film ["title"] not in friend_watched_movies:
+            unique_movies.append(film)
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies= []
+    users_watched_movies = get_users_watched_movie_titles(user_data)
+    
+    for movies in user_data["friends"]:
+        for movie in movies["watched"]:
+            if movie["title"] not in users_watched_movies\
+                and movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+    
+    return friends_unique_movies
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended_movies = []
+    users_watched_movies = get_users_watched_movie_titles(user_data)
+
+    for movies in user_data["friends"]:
+        for movie in movies["watched"]:
+            if movie["title"] not in users_watched_movies\
+            and movie["host"] in user_data["subscriptions"]:
+                recommended_movies.append(movie)
+
+    return recommended_movies
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    recommended_movies_by_genre = []
+    users_watched_movies = get_users_watched_movie_titles(user_data)
+
+    fav_genre = get_most_watched_genre(user_data)  
+
+    for movies in user_data["friends"]:
+        for movie in movies["watched"]:
+            if movie["title"] not in users_watched_movies\
+            and movie["genre"] == fav_genre:
+                recommended_movies_by_genre.append(movie)
+    
+    return recommended_movies_by_genre
+
+def get_rec_from_favorites (user_data): 
+    recommended_movies_from_favorites = []
+    friends_watched_movies = []
+    
+    for movies in user_data["friends"]:
+        for movie in movies["watched"]:
+            friends_watched_movies.append(movie["title"])

```

Comment:
I was a little surprised that you didn't make a helper method for this operation similar to `get_users_watched_movie_titles`!